File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        left, right = 0, len(arr) - 1

        while left <= right:
            mid = left + (right - left) // 2
            missing_number = arr[mid] - (mid + 1)

            if missing_number < k:
                left = mid + 1
            else:
                right = mid - 1

        return k + left

File: 1539-kth-missing-positive-number/1539-kth-missing-positive-number.rs
impl Solution {
    pub fn find_kth_positive(arr: Vec<i32>, k: i32) -> i32 {
        let mut result = Vec::new();
        
        for i in 1..=arr.last().unwrap() + k {
            if !arr.contains(&i){
                result.push(i);
            }
        }
        
    
        result[k as usize -1]
    }
    
}